app/pages/project/instances/NetworkingTab.tsx

```diff
@@ -363,6 +385,11 @@ export default function NetworkingTab() {
         },
       }
 
+      if (externalIp.kind === 'snat') {
+        // SNAT IPs can't be detached
+        return [copyAction]
+      }
+
```

Comment:
This is good, but it might be slightly nicer to leave the button in and disable it with "SNAT IPs can't be detached" as a hover message. Taking the opportunity to make clear this is on purpose.